providers/greenhouse.go

```diff
@@ -0,0 +1,43 @@
+package providers
+
+const Greenhouse Provider = "greenhouse"
+
+//nolint:lll
+func init() {
+	// Greenhouse Configuration
+	SetInfo(Greenhouse, ProviderInfo{
+		DisplayName: "Greenhouse",
+		AuthType:    Oauth2,
+		BaseURL:     "https://harvest.greenhouse.io",
+		Oauth2Opts: &Oauth2Opts{
+			GrantType:                 AuthorizationCode,
```

Comment:
Can you please add a comment in-code that OAuth 2.0 AuthCode is only applicable for Partner integrations, and not custom integrations? Please add a one-liner about partner vs custom integrations too